metrics/sources/kubelet/kubelet.go

```diff
@@ -132,25 +157,12 @@ func (this *kubeletMetricsSource) decodeMetrics(c *cadvisor.ContainerInfo) (stri
 			}
 		}
 
+		// No Kubernetes metadata so treat this as a system container.
 		if cName == "" || ns == "" || podName == "" {
```

Comment:
Yes, when everything works then everything has a name :).

But if you take a look couple lines above you will see that we sometimes extract the container name from an obfuscated docker container name and sometimes from a kubelet-assigned label. Both mechanisms can possibly fail and in that case I would rather drop the metric set/container with a warning instead of renaming it as a system.